src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs

```diff
@@ -54,13 +56,16 @@ internal sealed class CSharpSyntaxContext : SyntaxContext
             int position,
             SyntaxToken leftToken,
             SyntaxToken targetToken,
-            TypeDeclarationSyntax? containingTypeDeclaration,
-            BaseTypeDeclarationSyntax? containingTypeOrEnumDeclaration,
+            SyntaxNode? containingTypeDeclaration,
```

Comment:
can this be strengthened to BaseTypeDeclarationSyntax?

Reply:
No, because `containingTypeDeclaration` is used for both C# and VB in a shared `SyntaxContext` class

Reply:
but this is thr c# constructor :-)